pkg/operator/ceph/cluster/monitoring.go

```diff
@@ -0,0 +1,107 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cluster to manage a Ceph cluster.
+package cluster
+
+import (
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+)
+
+func (c *ClusterController) ConfigureCephMonitoring(cluster *cluster, cephUser string) {
+	var isEnabled bool
+	daemons := []string{"mon", "osd", "status"}
+
+	for _, daemon := range daemons {
+		// Is the monitoring enabled for that daemon?
+		isEnabled = isMonitoringConfigured(daemon, cluster.Spec)
+		if health, ok := cluster.monitoringChannels[daemon]; ok {
+			if health.monitoringRunning {
+				// If the goroutine was running but the CR was updated to stop the monitoring we need to close the channel
+				if isEnabled {
+					// close the channel so the goroutine can stop
+					close(cluster.monitoringChannels[daemon].stopChan)
+					// Set monitoring to false since it's not running anymore
+					cluster.monitoringChannels[daemon].monitoringRunning = false
+				} else {
+					logger.Debugf("ceph %s health go routine is already running for cluster %q", daemon, cluster.Namespace)
+				}
+			} else {
+				// if not already running and not disabled, we run it
+				if !isEnabled {
+					// Run the go routine
+					c.runGoroutine(cluster, daemon, cephUser)
+
+					// Set the flag that it is already running
+					cluster.monitoringChannels[daemon].monitoringRunning = true
+				}
+			}
+		} else {
+			// If the mon does not exist in the map, this is a first deployment or an operator restart
+			// So we check the desired state from the CR and run it if necessary
+			//
+			// If the mon monitoring is enabled
+			if !isEnabled {
+				cluster.monitoringChannels[daemon] = &clusterHealth{
+					stopChan:          make(chan struct{}),
+					monitoringRunning: false,
+				}
+
+				// Run the go routine
+				c.runGoroutine(cluster, daemon, cephUser)
+
+				// Set the flag that it is already running
+				cluster.monitoringChannels[daemon].monitoringRunning = true
+			}
+		}
+	}
+}
+
+func isMonitoringConfigured(daemon string, clusterSpec *cephv1.ClusterSpec) bool {
+	switch daemon {
+	case "mon":
+		return clusterSpec.HealthCheck.DaemonHealth.Monitor.Disabled
+
+	case "osd":
+		return clusterSpec.HealthCheck.DaemonHealth.ObjectStorageDaemon.Disabled
+
+	case "status":
+		return clusterSpec.HealthCheck.DaemonHealth.Status.Disabled
+	}
+
+	return false
+}
+
+func (c *ClusterController) runGoroutine(cluster *cluster, daemon string, cephUser string) {
```

Comment:
The prefix `start` is good for methods that start a goroutine
```suggestion
func (c *ClusterController) startMonitoringCheck(cluster *cluster, daemon string, cephUser string) {
```